x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -880,4 +805,32 @@ public static boolean isPlatinumOrTrialOperationMode(final OperationMode operati
     public synchronized XPackLicenseState copyCurrentLicenseState() {
         return new XPackLicenseState(this);
     }
+
+    private synchronized boolean isAllowedBySecurity() {
+        return isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
+    }
+
+    /**
+     * Test whether a feature is allowed by the status of current license and security config.
+     *
+     * @param minimumMode  The minimum license to meet or exceed
+     * @param needSecurity Whether security is required for feature to be allowed
+     * @param needActive   Whether current license needs to be active
+     * @param allowTrial   Whether the feature is allowed for trial license
+     *
+     * @return true if feature is allowed, otherwise false
+     */
+    private synchronized boolean isAllowedByLicenseAndSecurity(
+        OperationMode minimumMode, boolean needSecurity, boolean needActive, boolean allowTrial) {
+
+        final Status localStatus = status;
```

Comment:
I agree with your reasoning that since `status` is not volatile, we can remove the local variables from here and other places ( I don't think there was an explicit reason this was not done in #33396 )

Reply:
Removed. Also see my comments below